### PR TITLE
fix(ui): don't allow decor provider with ns_id==0

### LINKED ISSUE #19831: (UNKNOWN PLUGIN): Error executing lua: attempt to call a nil value
### Neovim version (nvim -v)

NVIM v0.8.0-dev-877-g35653e6bc

### Vim (not Nvim) behaves the same?

no

### Operating system/version

ArchLinux

### Terminal name/version

Alacritty

### $TERM environment variable

alacritty

### Installation

build from repo

### How to reproduce the issue

Since https://github.com/neovim/neovim/commit/35653e6bcda6923b5213fb9356c35067d6d0288f
`cat mini.lua`
```lua
local api = vim.api

local ns = api.nvim_create_namespace('test')
api.nvim_set_decoration_provider(ns, {
    on_start = function(...)
    end
})

_G.test = function()
    local bufnr = api.nvim_create_buf(false, true)
    local winid = api.nvim_open_win(bufnr, false, {
        col = -1,
        height = 5,
        relative = 'cursor',
        row = -1,
        width = 50
    })
    vim.wo[winid].winhl = 'Normal:Menu'
    api.nvim_buf_set_lines(bufnr, 0, -1, true, {'foo', 'bar'})
end

vim.defer_fn(function()
    _G.test()
end, 1000)
```
nvim --clean -u mini.lua mini.lua

### Expected behavior

No error. 

### Actual behavior

image

### PR DESCRIPTION
**Pull request by lewis6991**
Fixes #19831

### CODE REVIEW

**Comment by lewis6991** [comment]
Not sure if this is the right fix or not, but creating a decor_provider with `ns_id=0` seems to screw things up.

File: src/nvim/highlight.c [hunk 1]
```diff
@@ -174,7 +174,7 @@ int ns_get_hl(NS *ns_hl, int hl_id, bool link, bool nodefault)
 {
   static int recursive = 0;
 
-  if (*ns_hl < 0) {
+  if (*ns_hl <= 0) {
```
Comment (bfredl) [inline src/nvim/highlight.c:177]: `ns=0` should be allowed here. It means explicitly choosing the default namespace even if some other is active. But `get_decor_provider(ns_id, true);` and below should be skipped, as `ns=0` doesn't have a provider.

Comment (bfredl) [reply src/nvim/highlight.c:177]: ok, I think at line 183, adding `else if (ns_id == 0) { return -1; }` should be enough.

Comment (lewis6991) [reply src/nvim/highlight.c:177]: Ok that's actually what I tried first, so good to know I was in the right ballpark.

**Comment by bfredl** [comment]
cool. Would it be possible to write a test that triggers the error on master? perhaps from the case in https://github.com/neovim/neovim/issues/19831

**Comment by lewis6991** [comment]
A testcase for this would have made this much easier to debug and reverse engineer, unfortunately I was only able to reproduce this using my config and a precise step of actions, and even that didn't make the bug happen reliably.

I tried the testcase in #19831, but wasn't able to make it work. However, that testcase is along the right lines of setting up a decor provider and then setting `winhl`. What I found was that a decor provider with ns=0 wasn't just being created, but also executing callbacks, somehow, which suggests the provider's state was being currupted.

**Pull request merged**